### ISSUE TITLE
Activities synchronous with ISTEP

### ISSUE THREAD

**Issue by pjljvandelaar** [TorXakis/TorXakis#833]
The following TorXakis model (with line numbers)
```
 1. PROCDEF p [Input :: Int] () ::=
 2.        ISTEP
 3.    >-> Input ? x
 4.    >-> Input ? x | ISTEP
 5.    >-> ISTEP | Input ? x 
 6.    >-> STOP
 7. ENDDEF
 8.
 9. CHANDEF chans ::= Input :: Int ENDDEF
10. 
11. MODELDEF M ::
12.     CHAN IN Input
13.     CHAN OUT
14.     BEHAVIOUR
15.         p[Input]()
16. ENDDEF
```
should yields error messages on line 4 and 5, since
`ISTEP` can not be used synchronously with any activity .

When the model is offered to TorXakis the following error message is obtained:
```
PS C:\TorXakis.experiments> torxakis .\istep.txs

TXS >>  TorXakis :: Model-Based Testing

TXS >>  txsserver starting: "<a host name>" : 54019
TXS >>  Error {_errorType = ParseError, _errorLoc = ErrorLoc {errorLine = 5, errorColumn = 17}, _errorMsg = "(line 5, column 17):\nunexpected \"I\"\nexpecting \"||\", \"[\" or \"|\""}
TXS >>  CallStack (from HasCallStack):
TXS >>  error, called at src\TorXakis\Compiler.hs:168:21 in txs-compiler-0.1.0.0-71riZrtGetCYjOGtbMxYS:TorXakis.Compiler

TXS >>
```
The violation on line 4 is not detected:
The model with line 5 removed is accepted by the parser/compiler.

The problem is however more sincere that a parse error:
ISTEP is the empty set of observable actions.
Observable actions happen over channels.
So ISTEP can not be a channel.
Yet in [BehExprDefs.hs](https://github.com/TorXakis/TorXakis/blob/develop/sys/bexpr/src/BehExprDefs.hs#L263), it is defined as
```
chanIdIstep :: ChanId
chanIdIstep = ChanId "ISTEP" 902 []
```

**Comment by tretmans** [comment]
Cf. old compiler (V0.6.0):
```
TXS >>  TorXakis :: Model-Based Testing

TXS >>  txsserver starting: "PC-31093.tsn.tno.nl" : 51915
TXS >>
TXS >>  Parse Error: ( line = 5, column = 20 ) ISTEP
TXS >>  CallStack (from HasCallStack):
TXS >>  error, called at .stack-work\dist\67675594\build\TxsHappy.hs:3540:23 in front-0.2.1.0-BJqONMLGUVPJ1dKLZP8HJO:TxsHappy
TXS >>
```
There is a second issue here: the new compiler cannot deal with multi-actions not in { .. }.
If you change line 4 into   '{ Input ? x | ISTEP }'  and line 5 analogously the error above disappears.

**Comment by dnadales** [comment]
I can modify the parser so that multi actions not in `{ ... }` are accepted. However I think @tretmans and @pjljvandelaar should agree on what to do regarding the second issue: 

>The problem is however more sincere that a parse error:
ISTEP is the empty set of observable actions.
Observable actions happen over channels.
So ISTEP can not be a channel.

**Comment by pjljvandelaar** [comment]
closed by pr #843 
(remaining issue moved to #847)
